lib/core/versions/latest/TransactionSelector.ts

```diff
@@ -89,11 +93,14 @@ export default class TransactionSelector implements ITransactionSelector {
       && numberOfOperationsSeen < numberOfOperationsToQualify
       && transactionsPriorityQueue.length > 0) {
       const currentTransaction = transactionsPriorityQueue.pop();
-      const numOfOperationsInCurrentTransaction = AnchoredDataSerializer.deserialize(currentTransaction.anchorString).numberOfOperations;
-      numberOfOperationsSeen += numOfOperationsInCurrentTransaction;
-
-      if (numberOfOperationsSeen <= numberOfOperationsToQualify) {
-        transactionsToReturn.push(currentTransaction);
+      try {
+        const numOfOperationsInCurrentTransaction = AnchoredDataSerializer.deserialize(currentTransaction.anchorString).numberOfOperations;
+        numberOfOperationsSeen += numOfOperationsInCurrentTransaction;
+        if (numberOfOperationsSeen <= numberOfOperationsToQualify) {
+          transactionsToReturn.push(currentTransaction);
+        }
+      } catch {
+        console.debug(`Unable to deserialize ${currentTransaction.anchorString}, transaction not selected`);
```

Comment:
1. Please include the exception details (same as above).
1. Maybe we should change the message to be more generic as we are doing more things than just deserializing within try-catch. For example: An unexpected error occurred during processing; transaction is not selected. Error: ...